iOS/AppleSiliconSimulator.py

```diff
@@ -0,0 +1,182 @@
+    arm_node = intel_simulator_node
+    library_id_node = find_required_value(arm_node, 'LibraryIdentifier')
+    library_id_node.text = 'ios-arm64-simulator'
+    supported_archs_node = find_required_value(arm_node, 'SupportedArchitectures')
+    if supported_archs_node.tag != 'array':
+        raise Exception(f'Found unexpected tag in SupportedArchitectures array: {supported_archs_node.tag}')
+    for child in supported_archs_node:
+        if child.text.strip() == 'x86_64':
+            child.text = 'arm64'
+    return True
+
+def update_info_plist(path: str) -> bool:
+    '''
+    Update the Info.plist file for the XCFramework referenced by `path` to refer to an Arm64
+    Simulator instead of an Intel Simulator target. Returns `True` if the update was successful, or
+    `False` if the XCFramework has already been updated.
+    '''
+    info_plist_path = os.path.join(path, "Info.plist")
+    with open(info_plist_path) as input_file:
+        header = [next(input_file) for _ in range(2)]
+    tree = ET.parse(info_plist_path)
+    root = tree.getroot()
+    dict_node = root[0]
+    if dict_node.tag != 'dict':
+        raise Exception('First child in Info.plist is not dict!')
+    available_libraries = find_required_value(dict_node, 'AvailableLibraries')
+    if available_libraries.tag != 'array':
+        raise Exception(f'Found unexpected tag for AvailableLibraries array: {available_libraries.tag}')
+    if update_libraries_array(available_libraries):
+        tree.write(info_plist_path)
+        header_written = False
+        for line in fileinput.input(info_plist_path, inplace=True):
+            if not header_written:
+                header_written = True
+                for header_line in header:
+                    sys.stdout.write(header_line)
+            sys.stdout.write(line)
+        return True
+    return False
+
+def process_xcframework(path: str) -> None:
+    '''
+    Update the XCFramework referenced by `path` to contain an Arm64 Simulator instead of an Intel
+    Simulator.
+    '''
+    framework_name = os.path.basename(os.path.splitext(path)[0])
+    src_framework = os.path.join(path, 'ios-arm64', f'{framework_name}.framework')
+    src_path = os.path.join(src_framework, framework_name)
+    dst_framework = os.path.join(path, 'ios-arm64-simulator', f'{framework_name}.framework')
+    dst_path = os.path.join(dst_framework, framework_name)
+    minios_ver, sdk_ver = get_versions(src_path)
+    if update_info_plist(path):
+        copy_directory_tree(src_framework, dst_framework)
+        update_dst_framework(src_path, dst_path, minios_ver, sdk_ver)
+        print(f'{framework_name} updated.')
+    else:
+        raise Exception(f'{framework_name} has already been updated.')
+
+def main() -> None:
+    '''
+    The AppleSiliconSimulator.py main program.
+    '''
+    try:
+        if len(sys.argv) != 2:
+            raise Exception('Usage: AppleSiliconSimulator.py <Path to xcframework>')
```

Comment:
Maybe include the same info at the beginning of this file in the usage output?

Reply:
Fixed.

Reply:
Oops, I suggested the opposite of what you did. Having the usage at the top is good, but I'd like the summary be shown to the user too when they run the script with the incorrect number of args.

Reply:
OK, fixed.